src/core/raster/qgsrasterinterface.h

```diff
@@ -342,7 +342,7 @@ class CORE_EXPORT QgsRasterInterface
      * \param feedback optional feedback object
      */
     virtual QgsRasterBandStats bandStatistics( int bandNo,
-        int stats = QgsRasterBandStats::All,
+        int stats = static_cast< int >( QgsRasterBandStats::Statistic::All ),
```

Comment:
shouldn't we provide an alternate bandStatistics() helper that would accept QgsRasterBandStats::Statistic values to avoid the many int casts ?

Reply:
I think we could potentially just change this to the enum now without breaking the python API, but I'll need to confirm

Reply:
@rouault ok done, got it working nicely without the confusing ints in the API.

Reply:
@rouault I'm not going to be able to avoid static int casts in the next set, because those are for qt model roles which are always treated as into by qt API. (And we need to switch over an int which will have values from mixed enums too, ie Qt::ItemRole and the qgis side extensions to this.) . What's your thoughts on the operator+ approach used to avoid explicit static casts proposed in https://stackoverflow.com/a/42198760 ?

Reply:
> What's your thoughts on the operator+ approach used to avoid explicit static casts proposed in https://stackoverflow.com/a/42198760 ?

I'm discovering that there is a unary + operator in C++ :-) No strong opinion if we want this.  Perhaps ``int(x)`` would be a reasonable compromise?

Reply:
Do you mean a c style cast?